Store/rTorrentStateStore.cpp

```diff
@@ -130,13 +130,20 @@ bool rTorrentTorrentStateIterator::GetNext(Box& nextBox)
     std::unique_lock<std::mutex> lock(m_directoryItMutex);
 
     fs::path stateFilePath;
+    fs::path torrentFilePath;
+
     while (m_directoryIt != m_directoryEnd)
     {
         fs::path const& path = m_directoryIt->path();
         if (path.extension() == Detail::StateFileExtension && m_directoryIt->status().type() == fs::regular_file)
         {
             stateFilePath = path;
-            break;
+            torrentFilePath = stateFilePath;
+            torrentFilePath.replace_extension(fs::path());
+            if (fs::exists(torrentFilePath))
+            {
+                break;
```

Comment:
After the loop, there is a condition:

```cpp
     if (stateFilePath.empty())
     {
         return false;
     }
```

Let's adjust the code so that if we don't break the `stateFilePath` stays empty. Otherwise the error you're reporting could still take place.

Also, we may want to check `libTorrentStateFilePath` file existence here as well.